crates/lib/src/rules/convention/CV03.rs

```diff
@@ -0,0 +1,182 @@
+        SegmentSeekerCrawler::new(["select_clause"].into()).into()
+    }
+
+    fn eval(&self, rule_cx: RuleContext) -> Vec<LintResult> {
+        let segment = FunctionalContext::new(rule_cx.clone()).segment();
+        let children = segment.children(None);
+
+        let last_content: ErasedSegment =
+            children.clone().last().cloned().filter(|sp: &ErasedSegment| sp.is_code()).unwrap();
+
+        let mut fixes = Vec::new();
+
+        if self.select_clause_trailing_comma == "forbid" {
+            if last_content.is_type("comma") {
+                if last_content.get_position_marker().is_none() {
+                    fixes = vec![LintFix::delete(last_content.clone())];
+                } else {
+                    let comma_pos = last_content.get_position_marker().unwrap().source_position();
+
+                    for seg in rule_cx.segment.segments() {
+                        if seg.is_type("comma") {
+                            if seg.get_position_marker().is_none() {
+                                continue;
+                            }
+                        } else if seg.get_position_marker().unwrap().source_position() == comma_pos
+                        {
+                            if *seg != last_content.clone() {
+                                break;
+                            }
+                        } else {
+                            fixes = vec![LintFix::delete(last_content.clone())];
+                        }
+                    }
+                }
+
+                return vec![LintResult::new(
+                    Some(last_content),
+                    fixes,
+                    None,
+                    "Trailing comma in select statement forbidden".to_owned().into(),
+                    None,
+                )];
+            }
+        } else if self.select_clause_trailing_comma == "require" && !last_content.is_type("comma") {
+            let new_comma = SymbolSegment::create(",", &<_>::default(), <_>::default());
+
+            let fix: Vec<LintFix> = vec![LintFix::replace(
+                last_content.clone(),
+                vec![last_content.clone(), new_comma],
+                None,
+            )];
+
+            return vec![LintResult::new(
+                Some(last_content),
+                fix,
+                None,
+                "Trailing comma in select statement required".to_owned().into(),
+                None,
+            )];
+        }
+        Vec::new()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use pretty_assertions::assert_eq;
+
+    use crate::api::simple::{fix, lint};
+    use crate::core::rules::base::{Erased, ErasedRule};
+    use crate::rules::convention::CV03::RuleCV03;
+
+    fn rules() -> Vec<ErasedRule> {
+        rules_with_config("require".to_owned())
+    }
+
+    fn rules_with_config(select_clause: String) -> Vec<ErasedRule> {
+        vec![RuleCV03 { select_clause_trailing_comma: select_clause }.erased()]
+    }
```

Comment:
nit: Given we missed the type, I wonder if it is worth having a test that catches the "default case".